Common/Gfx/Canvas.cpp

```diff
@@ -563,6 +570,29 @@ void Canvas::DrawMaskedBitmap(Gdiplus::Bitmap* bitmap, Gdiplus::Bitmap* maskBitm
 	bitmapLock->Release();
 }
 
+void Canvas::DrawGeometry(const Shape & shape, D2D1_MATRIX_3X2_F & transform)
+{
+	if (!BeginTargetDraw()) return;
+
+	m_Target->SetAntialiasMode(D2D1_ANTIALIAS_MODE_ALIASED);
+	D2D1_MATRIX_3X2_F worldTransform;
+	m_Target->GetTransform(&worldTransform);
+	m_Target->SetTransform(transform * worldTransform);
+	Microsoft::WRL::ComPtr<ID2D1SolidColorBrush> solidBrush;
+	HRESULT hr = m_Target->CreateSolidColorBrush(shape.m_FillColor, solidBrush.GetAddressOf());
+	if (SUCCEEDED(hr)) {
+		if (shape.m_FillColor.a > 0)
+			m_Target->FillGeometry(shape.m_Shape.Get(), solidBrush.Get());
+
+		solidBrush->SetColor(shape.m_OutlineColor);
+		if (shape.m_OutlineColor.a > 0)
+			m_Target->DrawGeometry(shape.m_Shape.Get(), solidBrush.Get(), shape.m_OutlineWidth);
+	}
+	solidBrush.Reset();
```

Comment:
I had some stupid issues with ComPtrs and memory consumption, seems the gods of ComPtrs are kind today tho (I think it was because i was allocating a new brush on top of an existing. This would cause the previous ComPtr to never reset, right?)